project_issue_task_sync/README.rst

```diff
@@ -0,0 +1,35 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=====================
+Sync Issues and Tasks
+=====================
+
+This module syncs assigned user and state from task to issue and viceversa.
+If a task is created and does not have a issue a related issue will be
+generated. Projects may or may not have this syncing function, the activation
+of sync between issues and task may be activated on the project form by
+selecting the checkbox "sync issues and tasks". By default projects that use
+tasks and issues will be synced.
```

Comment:
Interesting historical perspective! thanks!
if you consider it like that, where an issue has multiple subtasks assigned to different people, this module  as it is does not make sense. The module is useful for projects where the project itself functions as "root task" and the "subtasks" are synced issues/tasks. It should be used only by users who have need for this usecase.
I can explain this better in the readme. 
i think that the root / substask structure could  work, where the root task is synced with the issue, but it is out of the scope for this idea.
In our usecase the issues and it's discussion are client facing. the discussions are not synced (sync_operations are not with messages). the tasks are work for the developers/consultants in the project.

in conclusion, yes "root task/ sub tasks" would accomplish the trick of syncing + having multiple subtasks for different users, but  we would be adding usecases and complexity: for example if i set the issue to "done", the root task should be synced to "done" even if it's subtasks are not completed? All these things should be decided and clear to the user before.
I think the syncing should be kept for what it is, an option for project managers to keep some of their projects with : (stage, assign to, name, description, project_id )   synced two ways. nothing more. 
I agree with how tasks an issues should work: tasks for backend and issues for frontend/helpdesk.